clar2wasm/src/standard/standard.wat

```diff
@@ -1409,29 +1409,81 @@
+        ;;      a. SHA-256 processes the data in 64-byte chunks, no matter how large the data is.
+        ;;      b. SHA-256 uses total 8 words (single word is of 4 bytes) as its initial values. These are pre-determined
+        ;;          1. In this module, from 0 index to 31 index, intial values are stored
+        ;;      c. SHA-256 uses 64 k-constants (single constant is of 4 bytes) to calculate final hash. These are also pre-determined
+        ;;          1. From 32 to 288, indices are reserved for k-constants
+        ;;      d. Take a look at initial values and k-constants here https://github.com/dandyvica/sha/blob/master/src/sha256.rs
+        ;;      g. SHA-256 requires its data to be in specific format (divisible by 64), before processing. 
+        ;;      h. SHA-256 performs 64 rounds (it's equal to the number of k-constants, and is also equal to total words calculated,
+        ;;         before calculating the actual hash)
+        ;; 3. To perform any calculation, sha256 requires the data to be in specific format. This process is called padding
+        ;;      a. Length of the padded data should always be divisible by 64 (i.e. $length%64==0), regardless of input size. 
+        ;;      b. For this purpose, it's essential to follow the following steps
+        ;;          1. Add '1' immediately after the last byte of the data
+        ;;          2. Leave last 8 bytes to store the length 
+        ;;          3. Fill the remaining space with zeros
+        ;;          4. Consider following 
+        ;;              a. if length=13 bytes, 1 byte to add after data, last 8 bytes for length, 64-(13+1+8)= 42 bytes of zeros
+        ;;              b. if length=123 bytes, 1 byte to add after data, last 8 bytes for length, 123+1+8 = 132, which is obviously 
+        ;;                 greater than 128, so next number which is divisible by 64 is 192. So 192-(123+1+8) = 60 bytes of zeros
+        ;;                 between 1 added after the data and last 8 bytes of length
+        ;;          5. Calculate length in bits (i.e. 13x8=104)
+        ;; 4. Before calculating hash and after padding sha256 data, algorithm needs to calculate total 64 words. 16 words are
+        ;;    obtained directly from the first 64 bytes of padded data. As each word is of 4 bytes, 16 words will require 64 bytes.
+        ;; 5. Next word will be calculated as w16 = w0 + σ0 + w9 + σ1 (w15 would be the 16th word as we are starting from 0). 
+        ;;    To take a look at how these are working, please visit https://sha256algorithm.com/
+        ;; 6. After calculating w63 (64th word), data is ready to perform hashing rounds.
+        ;; 7. Hashing rounds involve 8 variables, a,b,c,d,e,f,g and h. Each variable is of 4 bytes, hence sha256 will produce 32 bytes
+        ;;    result.
+        ;; 8. To take a look at how these words are being calculated, take a look at https://sha256algorithm.com
+        ;; 9. Finally result is being stored at specific location.
```

Comment:
I don't get the point of this. Anyone interested in rewriting this algorithm should already know it, or use the reference paper. Preferably both.

Reply:
I'm not assuming that next person(s) would be cryptographer, math lover or WASM ninja. I'm just saving their time as much as I can. I'm not assuming that they can't learn anything from papers or online resources, it's just that I want to make it easy for them.

Reply:
That's not a good assumption. This is a tough problem for motivated people, not an easy peasy little challenge for your average high level language dev.
People who can't read a scientific paper shouldn't be our choice of developer for cryptography functions.

Reply:
You've a good point there. I think purpose of these comments is to save time and effort, even if the person is brilliant. Some people might find paper easier after looking at the implementation. If we don't want that, then give me a green signal and I will remove it. Just a thought, would it be better if any other person gives opinion on that?